submissions/antonkottans/tjsw-post-oop/index.js

```diff
@@ -0,0 +1,185 @@
+const KEYS_VALUE_SEPARATOR = ": ";
+const PROPS_SEPARATOR = "; ";
+
+class Creature {
+    constructor({ name, gender, friends = [], species, saying }) {
+        this.name = name;
+        this.species = species;
+        this.gender = gender;
+        this.friends = friends;
+        this.saying = saying;
+    }
+
+    createArrayOfProps() {
+        return ["name", "species", "gender", "saying", "friends"]
+            .map((key) => {
+                if (key != "friends")
+                    return key + KEYS_VALUE_SEPARATOR + this[key];
+                return this.friends.length
+                    ? "friends: " +
+                          this.friends.map((friend) => friend.name).join(", ")
+                    : "";
+            })
+            .filter((_) => _);
```

Comment:
What filter would filter out?

Reply:
if there are no friends, one of results in map's return will be false

Reply:
Would it makes sense to strip friends **before** they are processed?